### ISSUE TITLE
[Feature] Cannot create a version with build metadata

### ISSUE THREAD

**Issue by tyb-dev** [GitTools/GitVersion#3425]
**Describe the bug**

## Expected Behavior
Create a build with SemVer2 build metadata attached which is specific for a particular release. 

## Actual Behavior
According to SemVer2, the build metadata may be used to add additional information to a release. GitVersion ignores build metadata in both `next-version` and `git tag`. 

## Possible Fix
When a release tag explicitly includes build metadata, include it in the version string, or allow specifying it via branch name or commit message. 

## Steps to Reproduce
```
$ tag 1.0.0+special
$ gitversion
(...)
 "FullSemVer": "1.0.0",
```

## Context
I'd like to create special builds that sometimes come with build metadata.

## Your Environment
Ubuntu 22.04

**Comment by arturcic** [comment]
@asbjornu

**Comment by asbjornu** [comment]
Yeah, this seems like a hole we should fill. Pull requests welcome!

**Comment by arturcic** [comment]
@tyb-dev  can you tell us which version did you use ? and also if you have tried the latest beta? https://www.nuget.org/packages/GitVersion.Tool/6.0.0-beta.1
